console-subscriber/src/visitors.rs

```diff
@@ -0,0 +1,359 @@
+                if p0 == Self::OP_STATE_FIELD_PREFIX {
+                    if p2 == Self::OP_STATE_FIELD_TYPE_UNIT {
+                        return Some(MatchPart {
+                            attr_name: p1.into(),
+                            part_type: MatchPartType::Unit(value.into()),
+                        });
+                    } else if p2 == Self::OP_STATE_FIELD_TYPE_VALUE {
+                        return Some(MatchPart {
+                            attr_name: p1.into(),
+                            part_type: MatchPartType::Text(value.into()),
+                        });
+                    } else if p2 == Self::OP_STATE_FIELD_TYPE_OP {
+                        return Some(MatchPart {
+                            attr_name: p1.into(),
+                            part_type: MatchPartType::OpType(value.into()),
+                        });
+                    }
+                }
+            }
+            None
+        };
+
+        if field.name() == Self::OP_NAME_FIELD_NAME {
+            self.op_name = Some(value.to_string());
+            return;
+        } else if field.name() == Self::OP_TYPE_FIELD_NAME {
+            self.op_type = Some(value.to_string());
+            return;
+        } else if field.name() == Self::OP_READINESS_FIELD_NAME {
+            self.readiness = Some(match value {
+                Self::OP_READINESS_READY => Readiness::Ready,
+                Self::OP_READINESS_PENDING => Readiness::Pending,
+                _ => return,
+            });
+            return;
+        }
+
+        if let Some(match_part) = extract_partial_attr() {
+            match match_part.part_type {
+                MatchPartType::Text(t) => {
+                    self.state_text_attrs.insert(match_part.attr_name, t);
+                }
+                MatchPartType::OpType(op_type) => {
+                    let attr = self
+                        .state_numeric_attrs
+                        .entry(match_part.attr_name)
+                        .or_default();
+                    attr.op = Some(op_type);
+                }
+                MatchPartType::Unit(unit) => {
+                    let attr = self
+                        .state_numeric_attrs
+                        .entry(match_part.attr_name)
+                        .or_default();
+                    attr.unit = Some(unit);
+                }
+            }
+        }
+    }
+
+    fn record_u64(&mut self, field: &tracing_core::Field, value: u64) {
+        let mut parts = field.name().split('_');
+        let p0 = parts.next();
+        let p1 = parts.next();
+        let p2 = parts.next();
+        if let (Some(p0), Some(p1), Some(p2)) = (p0, p1, p2) {
+            if p0 == Self::OP_STATE_FIELD_PREFIX && p2 == Self::OP_STATE_FIELD_TYPE_VALUE {
+                let attr = self.state_numeric_attrs.entry(p1.into()).or_default();
+                attr.val = Some(value)
+            }
+        }
```

Comment:
i think this can be simplified to
```suggestion
        let parts = (parts.next(), parts.next(), parts.next());
        if let (
            Some(Self::OP_STATE_FIELD_PREFIX),
            Some(p1),
            Some(Self::OP_STATE_FIELD_TYPE_VALUE),
        ) = parts {
            let attr = self.state_numeric_attrs.entry(p1.into()).or_default();
            attr.val = Some(value)
        }
```